src/main/java/use_case/search/SearchInteractor.java

```diff
@@ -2,20 +2,79 @@
+import org.json.JSONArray;
+import org.json.JSONObject;
+import entity.University;
+import entity.UniversityFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
 
 public class SearchInteractor implements SearchInputBoundary {
     final SearchUserDataAccessInterface searchDataAccessObject;
     final SearchOutputBoundary searchPresenter;
+    final UniversityFactory universityFactory;
 
-    public SearchInteractor(SearchUserDataAccessInterface signupDataAccessInterface,
-                            SearchOutputBoundary signupOutputBoundary) {
-        this.searchDataAccessObject = signupDataAccessInterface;
+    public SearchInteractor(SearchUserDataAccessInterface searchUserDataAccessInterface,
+                            SearchOutputBoundary signupOutputBoundary, UniversityFactory universityFactory) {
+        this.searchDataAccessObject = searchUserDataAccessInterface;
         this.searchPresenter = signupOutputBoundary;
+        this.universityFactory = universityFactory;
     }
 
     @Override
     public void execute(SearchInputData searchInputData) {
+        String stringAccumulator = "";
+        String input = searchInputData.getSearchCriteria();
+        for (int i = 0; i < input.length(); i++) {
+            String cursor = input.substring(i, i+1);
+            if (cursor.equals(" ")) {
+                stringAccumulator += "%20";
+            }
+            else {
+                stringAccumulator += cursor;
+            }
+        }
+        JSONObject query = searchDataAccessObject.searchQuery("school.name=" + stringAccumulator);
+        JSONObject metadata = query.getJSONObject("metadata");
+        if (metadata.getInt("total") == 0) {
+            searchPresenter.prepareResultsNotFoundView("Results not found!");
+        }
+        else {
+            List<University> universityList = executeHelper(query.getJSONArray("results"));
+            SearchOutputData searchOutputData = new SearchOutputData(universityList);
+            searchPresenter.prepareSuccessView(searchOutputData);
+        }
+    }
+
+    private List<University> executeHelper(JSONArray results) {
+        List<University> universities = new ArrayList<University>();
+        for (int i = 0; i < results.length(); i++) {
+            JSONObject university = (JSONObject) results.get(i);
+            Integer ID = university.getInt("id");
+            String name = university.getString("school.name");
+            String state = university.getString("school.state");
+            String city = university.getString("school.city");
+            Object admRate = university.get("admissions.admission_rate.overall");
+            // Need to change object to float and accept null values
+            if (admRate == "null") {
+                Double admRateDouble = null;
+            }
+            Float admRateConvert = (Float) admRate;
+            Double admRateDouble = admRateConvert.doubleValue();
```

Comment:
Need to add some additional type handling here to deal with when one of the requested fields is Null (i.e no data is available). Otherwise, we run into and error, for instance: https://api.data.gov/ed/collegescorecard/v1/schools.json?school.name=Harvard&fields=id,school.name,school.state,school.city,admissions.admission_rate.overall,cost.tuition.in_state,cost.tuition.out_of_state,admissions.sat_scores.average.overall,admissions.act_scores.midpoint.cumulative,school.school_url&api_key=AwypPp5bIHx6H4EevsB5sFs4gw5NK1M7SNc4eXwl